src/omitBy.ts

```diff
@@ -0,0 +1,46 @@
+import { purry } from './purry';
+import { Key } from './_types';
+
+/**
+ * Returns a partial copy of an object omitting the keys matching predicate.
+ * @param object the target object
+ * @param fn the predicate
+ * @signature R.omitBy(object, fn)
+ * @example
+ *    R.omitBy({a: 1, b: 2, A: 3, B: 4}, (val, key) => key.toUpperCase() === key) // => {a: 1, b: 2}
+ * @data_first
+ * @category Object
+ */
+export function omitBy<K extends Key, V>(
+  object: Record<K, V>,
+  fn: (value: V, key: K) => boolean
+): Record<K, V>;
+
+/**
+ * Returns a partial copy of an object omitting the keys matching predicate.
+ * @param fn the predicate
+ * @signature R.omitBy(fn)(object)
+ * @example
+ *    R.omitBy((val, key) => key.toUpperCase() === key)({a: 1, b: 2, A: 3, B: 4}) // => {a: 1, b: 2}
+ * @data_last
+ * @category Object
+ */
+export function omitBy<K extends Key, V>(
+  fn: (value: V, key: K) => boolean
+): (object: Record<K, V>) => Record<K, V>;
+
+export function omitBy() {
+  return purry(_omitBy, arguments);
+}
+
+function _omitBy(object: any, fn: (value: any, key: any) => boolean) {
+  if (object == null) {
+    return {};
+  }
```

Comment:
I think it's pretty weird that `omitBy` and `pickBy` both accept `undefined` or `null` as input, when the types clearly state that you cannot pass that. It shows in the tests because we need `as any`.

Other methods like `map` or `filter` do _not_ allow passing nullish values, and I think this is on purpose, given the discussion here:
- https://github.com/remeda/remeda/issues/147

It seems that `pick` also does a null check, but `omit` does not 🤔 

---

so my thoughts here would be to:
- align `omitBy` with `omit` rather than with `pickBy` and remove this null check
- it was added to `pick` because of:
  - https://github.com/remeda/remeda/issues/4

but given that it is the _only_ function that does that, we should consider removing this check in `pick` and `pickBy` as well, either in the next major version (too bad we see this right after v1, lol), or, just in a minor because one goal is to align the implementation with the types... what do you think?

Reply:
Hard question :)

I have some options:
- We can add nullckecks to every function, think this feature could be useful. But we should update types.
- We can remove all nullcheck, make v2, coz this is breaking change. I think that can help us to slightly improve performance (so slightly that we won’t mention) but make lib harder to use.
- Remove this check from omitBy for now, and discuss this problem later :)

Reply:
First option not ultimate solution too. We should discuss some points. For example, which type should be returned - null/undefined (as on input) or empty array/object.

Reply:
Second option improves transparency: if input was incorrect we'll throw error instead of doing other's job

Reply:
Third option solves nothing but rising tech debt

Reply:
Removed this check here for now

Reply:
yeah let's just keep this in mind for the future: no more unnecessary null checks, and maybe we can remove the existing ones some day :)